src/melee.cpp

```diff
@@ -1870,11 +1870,11 @@ std::vector<special_attack> player::mutation_attacks(Creature &t) const
             }
 
             ///\EFFECT_UNARMED increases chance of attacking with mutated body parts
-
             ///\EFFECT_DEX increases chance of attacking with mutated body parts
-            const int chance = std::max( 1, mut_atk.chance - get_dex() - unarmed );
+            const float chance = get_dex() + unarmed / mut_atk.chance;
             add_msg( m_debug, "%s proc chance: %d", pr.first.c_str(), chance );
-            if( chance != 1 && !one_in( chance ) ) {
+            // If the mutation attack can fail to proc, and it does fail to proc, bail out
+            if( get_dex() + unarmed < mut_atk.chance && !x_in_y( get_dex() + unarmed, mut_atk.chance ) ) {
```

Comment:
That was because in the former code, the chance was expected to go down to 1.

Going by your description:
> Since chance scales linearly rather than exponentially, mutation attacks can now be made very uncommon without fear of them never actually showing up no matter how high your skill is.

It looks like you want to make it so that the chance generally doesn't reach 1, thus making the special case redundant.

There also was the problem of code quality: old code had a simple `!= 1`, while the new one had the chance calculations repeated 3 times (for debugmsg, for special case and finally for x_in_y).